File: frontend/src/types/portfolio.ts
// backend/app/schemas/portfolio.py 1:1 대응 (docs/02-coding-conventions.md 9장)

export interface PortfolioSummary {
  krw_balance: string;
  coin_valuation: string;
  net_deposit: string;
}

export interface CoinRef {
  symbol: string;
  korean_name: string;
}

export interface HoldingItem {
  coin_symbol: string;
  korean_name: string;
  quantity: string;
  avg_buy_price: string;
  current_price: string;
  valuation: string;
  profit: string;
  profit_pct: string;
}

export type TradeSide = "buy" | "sell";
export type TradeSource = "manual" | "auto";

export interface TradeItem {
  id: number;
  coin_symbol: string;
  korean_name: string;
  side: TradeSide;
  source: TradeSource;
  price: string;
  quantity: string;
  filled_at: string;
}

export interface TradeListResponse {
  items: TradeItem[];
  total: number;
  traded_coins: CoinRef[];
}

export interface MonthlyProfit {
  month: string;
  profit: string;
}

export type ReportPeriod = "1m" | "3m" | "6m" | "1y" | "all";

export interface PortfolioReport {
  start: string | null;
  end: string;
  most_traded_coin: CoinRef | null;
  monthly_profits: MonthlyProfit[];
}

File: backend/app/schemas/portfolio.py
"""08-portfolio 응답 DTO.

금액·수량 필드는 문자열로 반환한다 (02-coding-conventions.md 9절).
"""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class PortfolioSummaryResponse(BaseModel):
    krw_balance: str
    coin_valuation: str
    net_deposit: str


class CoinRef(BaseModel):
    symbol: str
    korean_name: str


class HoldingItem(BaseModel):
    coin_symbol: str
    korean_name: str
    quantity: str
    avg_buy_price: str
    current_price: str
    valuation: str
    profit: str
    profit_pct: str


class TradeItem(BaseModel):
    id: int
    coin_symbol: str
    korean_name: str
    side: Literal["buy", "sell"]
    source: Literal["manual", "auto"]
    price: str
    quantity: str
    filled_at: datetime


class TradeListResponse(BaseModel):
    items: list[TradeItem]
    total: int
    traded_coins: list[CoinRef]


class MonthlyProfit(BaseModel):
    month: str
    profit: str


class PortfolioReportResponse(BaseModel):
    start: datetime | None
    end: datetime
    most_traded_coin: CoinRef | None
    monthly_profits: list[MonthlyProfit]
